src/App.tsx

```diff
@@ -1,223 +1,175 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Loader } from './components/Loader';
+import { TodoList } from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [todoIdsToRemove, setTodoIdsToRemove] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo>({
+    id: 0,
+    userId: 0,
+    title: '',
+    completed: false,
+  });
+
+  const getTodosFromServer = useCallback(async () => {
+    try {
+      if (user) {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      }
+    } catch (error) {
+      setHasError(true);
+      setErrorMessage('Can`t download ToDos from server');
+    }
+  }, []);
+
+  const addNewTodoToServer = useCallback(async (title: string) => {
+    try {
+      if (user) {
+        setIsAdding(true);
+
+        const currTodo = {
+          title,
+          userId: user.id,
+          completed: false,
+        };
+
+        setTempTodo(prev => ({ ...prev, currTodo }));
+
+        await addTodo(currTodo);
+        await getTodosFromServer();
```

Comment:
No need to send this request, you can get new todo from `addTodo` method, and then add it to the list, why to send additional request every time you adding new todo